pinot-common/src/main/java/org/apache/pinot/common/function/scalar/DateTimeConvert.java

```diff
@@ -42,36 +51,43 @@ public Object dateTimeConvert(String timeValueStr, String inputFormatStr, String
       _inputFormatSpec = new DateTimeFormatSpec(inputFormatStr);
       _outputFormatSpec = new DateTimeFormatSpec(outputFormatStr);
       _granularitySpec = new DateTimeGranularitySpec(outputGranularityStr);
+      _dateTime = new MutableDateTime(0L, DateTimeZone.UTC);
+      _buffer = new StringBuilder();
```

Comment:
Consider adding some javadoc explaining we reuse date time and string builder to reduce garbage

Reply:
As an implementation detail, I don't think it should go into javadoc. If needed I can add some regular comments though. 
I think it'd be more effective to go through codebase and fix places that utilize joda-time inefficiently (e.g. in DateTimeFunctions class) to eradicate bad pattern(s).

Reply:
Sorry I meant comments instead of javadoc. The purpose is to educate future developers

Reply:
Added in latest commit.